### ISSUE TITLE
bug: java.util.concurrent.TimeoutException

### ISSUE THREAD

**Issue by dominiccooney** [sourcegraph/jetbrains#1694]
### Cody Version

6.0.0-nightly

### IDE Information

IntelliJ IDEA 2024.1.1 (Ultimate Edition)
Build #IU-241.15989.150, built on April 29, 2024
Licensed to Sourcegraph / Dominic Cooney
Subscription is active until February 28, 2025.
Runtime version: 17.0.10+1-b1207.14 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 14.4.1
Kotlin analyzer version: 2.0.0-ij241.15989-9
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 12
Metal Rendering is ON
Registry:
  ide.experimental.ui=true
  markdown.open.vcs.log.link=true
  llm.generate.commit.summary=true
Non-Bundled Plugins:
  com.jetbrains.space (241.15989.9)
  google-java-format (1.22.0.0)
  DevKit (241.15989.199)
  io.intellij-sdk-thread-access (2.2.1)
  com.intellij.ml.llm (241.15989.156)
  com.jetbrains.packagesearch.intellij-plugin (241.0.9)
  com.sourcegraph.jetbrains (6.0.0-nightly)
Kotlin: 241.15989.150-IJ

### Describe the bug

Repro step:

* Open a second project in a new window.

### Expected behavior

Actual behavior: Cody Agent won't start. Agent is working fine in the first window though.

Expected behavior: I can use Cody in multiple windows simultaneously.

### Additional context

Stacktrace:

```text
java.util.concurrent.TimeoutException
 at java.base/java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1960)
 at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2095)
 at com.sourcegraph.cody.agent.CodyAgentService.startAgent$lambda$4(CodyAgentService.kt:181)
 at com.intellij.openapi.application.impl.RwLockHolder$executeOnPooledThread$1.run(RwLockHolder.kt:154)
 at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
 at com.intellij.util.concurrency.ContextCallable.call(ContextCallable.java:32)
 at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
 at com.intellij.util.concurrency.ContextRunnable.run(ContextRunnable.java:27)
 at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
 at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
 at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
 at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
 at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
 at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
 at java.base/java.lang.Thread.run(Thread.java:840)
 
```

**Comment by dominiccooney** [comment]
Closing the first window with the working Cody has now hung the second window... I'm beachballin'.

**Comment by dominiccooney** [comment]
Tried again, and this time both status bars say "Cody encountered an unexpected error" but the first window is responding to chat.

**Comment by odisseus** [comment]
Fixed by #1706.
